### PR TITLE
feat(providers): add SimpleLogin oidc

### PR DESCRIPTION
**Pull request by alexaka1**
## ☕️ Reasoning

Adds *Sign in with SimpleLogin* support. 

> SimpleLogin is built for privacy-conscious site owners and protects your users identity. Open-source, Developer-first, SimpleLogin can be integrated in minutes.

## 🧢 Checklist

- [X] Documentation
- [X] Tests
- [X] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)
- [Sign in with SimpleLogin](https://simplelogin.io/developer/)

### CODE REVIEW

**Comment by alexaka1** [comment]
I am not sure what tests are necessary. I have tested the code with my own website, and users are able to log in with SimpleLogin. I would love some input from maintainer about tests.

**Comment by ndom91** [comment]
This looks pretty good already! Can you add a docs page at `/docs/pages/getting-started/providers/simplelogin.mdx`? 

Thanks 🙏

**Comment by alexaka1** [comment]
I have included an `mdx`.

**Comment by ndom91** [comment]
Two minor nit picks,

1. Is the brand color really `#fff`? Maybe lets go with their pink-ish color from the logo
2. There seems to be a prettier issue in your mdx file (`docs/pages/getting-started/providers/simplelogin.mdx`)

**Comment by alexaka1** [comment]
Tbh, I couldn't figure out what the brand color does. But will get them fixed.

**Review by ndom91 — APPROVED** [review]

File: packages/core/src/providers/simplelogin.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+ *  - [Sign in with SimpleLogin](https://simplelogin.io/developer/)
+ *  - [SimpleLogin OAuth documentation](https://simplelogin.io/docs/siwsl/intro/)
+ *  - [SimpleLogin OAuth Configuration](https://app.simplelogin.io/developer)
+ *
+ * ### Notes
+ *
+ * By default, Auth.js assumes that the SimpleLogin provider is
+ * based on the [Open ID Connect](https://openid.net/specs/openid-connect-core-1_0.html) specification.
+ *
+ *
```
Comment (k-taro56) [inline packages/core/src/providers/simplelogin.ts:55]: New line duplicated
```suggestion
```

File: packages/core/src/providers/simplelogin.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+ * #### Configuration
+ *```js
+ * import Auth from "@auth/core"
+ * import SimpleLogin from "@auth/core/providers/simplelogin"
+ *
+ * const request = new Request(origin)
+ * const response = await Auth(request, {
+ *   providers: [SimpleLogin({ clientId: SIMPLELOGIN_CLIENT_ID, clientSecret: SIMPLELOGIN_CLIENT_SECRET })],
+ * })
```
Comment (k-taro56) [inline packages/core/src/providers/simplelogin.ts:41]: `pnpm format`
```suggestion
 * const response = await Auth(request, {
 *   providers: [
 *     SimpleLogin({
 *       clientId: SIMPLELOGIN_CLIENT_ID,
 *       clientSecret: SIMPLELOGIN_CLIENT_SECRET,
 *     }),
 *   ],
 * })
```

File: packages/core/src/providers/simplelogin.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+ *
+ * ### Setup
+ *
+ * #### Callback URL
+ * ```
+ * https://example.com/api/auth/callback/simplelogin
+ * ```
+ *
+ * #### Configuration
+ *```js
```
Comment (k-taro56) [inline packages/core/src/providers/simplelogin.ts:34]: `js` to `ts`
```suggestion
 *```ts
```

File: packages/core/src/providers/simplelogin.ts [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    issuer: "https://app.simplelogin.io",
+    profile(profile) {
+      return {
+        id: profile.sub,
+        name: profile.name,
+        email: profile.email,
+        image: profile.avatar_url,
+      }
+    },
+    style: { logo: "/simplelogin.svg", brandColor: "#e3156a" },
```
Comment (ndom91) [inline packages/core/src/providers/simplelogin.ts:104]: Not necessary anymore

```suggestion
    style: { brandColor: "#e3156a" },
```

**Pull request merged**